lib/opal/rewriters/logical_operator_assignment.rb

```diff
@@ -0,0 +1,199 @@
+require 'opal/rewriters/base'
+
+module Opal
+  module Rewriters
+    class LogicalOperatorAssignment < Base
+      class AbstractHandler < self
+        def self.get(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.set(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.process(lhs, rhs, root_type)
+          s(root_type,
+            get(lhs, rhs),
+            set(lhs, rhs))
+        end
+
+        def self.new_temp
+          @@counter ||= 0
+          @@counter += 1
+          :"$or_equals_lhs_recvr_tmp_#{@@counter}"
```

Comment:
@iliabylich this seems a bit too long, maybe `$oreql_lhs_#{@@counter}`?

Also I wouldn't trust much Opal implementation of cvars… 😏